### DISCUSSION TITLE
Is there a possibility of novint falcon haptic device implantation

### DISCUSSION THREAD

**Discussion by JauJieYou** [WPI-AIM/ambf#198] [Q&A]
Is there a possibility of novint falcon haptic device implantation?

**Comment by adnanmunawar** [comment]
Hello, 
We do support the Novint Falcon device. What exactly are you looking to do?

**Reply by JauJieYou** [reply]
Dear Sir:
  Thanks for your reply. I am looking for the purpose, of using two Novint Falcon, as input control as ECM, to  [AMBF simulation](https://github.com/collaborative-robotics/surgical_robotics_challenge).  Is it possible? I only can find for [Phantom Omni](https://github.com/WPI-AIM/ros_geomagic), and the ['scripts/surgical_robotics_challenge/geomagic_multi_psm_control.py' file](https://github.com/collaborative-robotics/surgical_robotics_challenge/tree/master/scripts/surgical_robotics_challenge) is missing?
  And should I install other program, like [libnifalcon](https://github.com/libnifalcon), or [sawForceDimensionSDK](https://github.com/jhu-saw/sawForceDimensionSDK), or [Force Dimension SDK](https://www.forcedimension.com/software)?
  I have no budget to using more expensive device. But I will buy two falcon if it is possible.

**Reply by adnanmunawar** [reply]
Hello,
I see, are you looking to control the ECM in the surgical robotics challenge? If so then such a script should be easy to create. And yes, you will need to install `libnifalcon` as part of this package (https://github.com/WPI-AIM/ros_falcon). This ros_falcon package can then be run to publish the pose of the Novint Falcon via ROS topics. After this you can create a script based on this one (https://github.com/collaborative-robotics/surgical_robotics_challenge/blob/master/scripts/surgical_robotics_challenge/examples/ecm_control.py) to subscribe to the Novint Falcon's pose and move the ECM.

**Comment by JauJieYou** [comment]
It's great.
  May I ask another question? In your experience, can I use two falcon, as
bilateral hands?
   And another question is that the falcon is 3 DoF,  can it rotate?
Because the needle looks rotate while suturing.
  Thanks.

Best Wishes,
JJ You

Adnan Munawar ***@***.***> 於 2023年4月15日 週六 上午12:17 寫道：

**Reply by adnanmunawar** [reply]
Hello,

I think it should be possible to use two Falcons at once.

The falcon, as you noticed, has only 3 DOF so not it would not be able to control the gripper orientation.

**Comment by JauJieYou** [comment]
Thank you for the clarification. But I have no money for the sigma.7, I had
tried to ask the cost, and scared.

Thanks

Best Wishes,
JJ You

Adnan Munawar ***@***.***> 於 2023年5月11日 週四 上午1:57 寫道：

**Comment by adnanmunawar** [comment]
Yes you don't need to buy a Simga. You can instead look at Geomagic Touch or Phantom Omni. The Phantom Omni should be inexpensive if you can find it on lets say Ebay.

Or you can just a SpaceNav mouse which will allow to you control the 6 DOFs
